.github/run-integration-test.sh

```diff
@@ -9,67 +9,106 @@ set -Euo pipefail
 # - COSMOSDB_ACCOUNT_NAME
 
 # Azure Storage Blob Extension
-export QUARKUS_AZURE_STORAGE_BLOB_CONNECTION_STRING=$(az storage account show-connection-string \
-  --resource-group "${RESOURCE_GROUP_NAME}" \
-  --name "${STORAGE_ACCOUNT_NAME}" \
-  --query connectionString -o tsv)
+# Authenticate to Azure Storage Blob with Microsoft Entra ID and connection string
+# Get the endpoint of azure storage blob
+AZURE_STORAGE_BLOB_ENDPOINT=$(az storage account show \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --name $STORAGE_ACCOUNT_NAME \
+    --query 'primaryEndpoints.blob' \
+    --output tsv)
+# Retrieve the storage account resource ID
+STORAGE_ACCOUNT_RESOURCE_ID=$(az storage account show \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --name $STORAGE_ACCOUNT_NAME \
+    --query 'id' \
+    --output tsv)
+# Assign the "Storage Blob Data Contributor" role to the current signed-in identity
+servicePrincipal=$(az ad sp list --filter "appId eq '$AZURE_CLIENT_ID'" --query '[0].id' -o tsv)
+az role assignment create \
+    --assignee ${servicePrincipal} \
+    --role "Storage Blob Data Contributor" \
+    --scope $STORAGE_ACCOUNT_RESOURCE_ID
+# Get the connection string that has full access to the account
+AZURE_STORAGE_BLOB_CONNECTION_STRING=$(az storage account show-connection-string \
+    --resource-group "${RESOURCE_GROUP_NAME}" \
+    --name "${STORAGE_ACCOUNT_NAME}" \
+    --query connectionString -o tsv)
+
+# Run integration test with existing native executables against Azure services
+export QUARKUS_AZURE_STORAGE_BLOB_ENDPOINT=${AZURE_STORAGE_BLOB_ENDPOINT}
+unset QUARKUS_AZURE_STORAGE_BLOB_CONNECTION_STRING
+mvn -f azure-storage-blob/pom.xml -B test-compile failsafe:integration-test -Dnative -Dazure.test=true
+export QUARKUS_AZURE_STORAGE_BLOB_CONNECTION_STRING=${AZURE_STORAGE_BLOB_CONNECTION_STRING}
+unset QUARKUS_AZURE_STORAGE_BLOB_ENDPOINT
+mvn -f azure-storage-blob/pom.xml -B test-compile failsafe:integration-test -Dnative -Dazure.test=true
+
+# Run both unit test and integration test in JVM mode against Azure services
+export QUARKUS_AZURE_STORAGE_BLOB_ENDPOINT=${AZURE_STORAGE_BLOB_ENDPOINT}
+unset QUARKUS_AZURE_STORAGE_BLOB_CONNECTION_STRING
+mvn -f azure-storage-blob/pom.xml -B verify -Dazure.test=true
+export QUARKUS_AZURE_STORAGE_BLOB_CONNECTION_STRING=${AZURE_STORAGE_BLOB_CONNECTION_STRING}
+unset QUARKUS_AZURE_STORAGE_BLOB_ENDPOINT
+mvn -f azure-storage-blob/pom.xml -B verify -Dazure.test=true
```

Comment:
how about 
```
mvn -f azure-storage-blob/pom.xml -B verify -Dazure.test=true -Dquarkus.azure.storage.blob.connection-string=${AZURE_STORAGE_BLOB_CONNECTION_STRING}
```

Reply:
Nice suggestion, thanks! Fixed in 428f8edd575966a6fd77a9cc62bac89f3421d0f5.